R/score_plates.R

```diff
@@ -266,7 +264,7 @@ optimize_multi_plate_design <- function(batch_container, across_plates_variables
       if (!quiet && length(plate_levels) > 1) cat(curr_plate, "... ")
 
       bc <- optimize_design(bc,
-        scoring = scoring_funcs,
+        scoring = scoring_funcs %>% purrr::map(curr_plate) %>% rlang::set_names(paste(names(.), " Plate_", curr_plate)),
```

Comment:
not sure I understand this code. 

`scoring_funcs` is a list of lists now, and `curr_plate` is a a plate id (character or numeric). right?

what does `scoring_funcs %>% purrr::map(curr_plate)` do?

maybe we need to transpose the `scoring_funcs` first with `purrr::list_transpose`, and then we:

1. `assert_that` length of `plate_levels` is the same of `scoring_funcs`?
2. `for (plate_index in seq_len(length(plate_levels)`

Reply:
> not sure I understand this code. 
> 
> `scoring_funcs` is a list of lists now, and `curr_plate` is a a plate id (character or numeric). right?
> 
> what does `scoring_funcs %>% purrr::map(curr_plate)` do?

First part is correct.
`purrr::map(curr_plate)` is equivalent to `purrr::map(purrr::pluck, curr_plate)`
Maybe it should be written like that for readability.

> maybe we need to transpose the `scoring_funcs` first with `purrr::list_transpose`, and then we:
> 
> 1. `assert_that` length of `plate_levels` is the same of `scoring_funcs`?
> 2. `for (plate_index in seq_len(length(plate_levels)`

I need to check this.